File: problem_set/1073_b_vasya_and_books.py
'''
  Contest : 1073
  Problem B : Vasya and books
  Author : Rakesh Kumar
  Date: 12/07/2020
'''

def solve(books):
    deck = list(map(int, input().split(' ')))
    top = 1
    for b in deck:
        if books[b] >= top:
            print(books[b] - top + 1, end=' ')
            top = books[b] + 1
        else:
            print('0', end=' ')
    print()

if __name__ == '__main__':
    n = int(input())
    temp = list(map(int, input().split(' ')))
    books = {}
    for i in range(1, n + 1):
        books[temp[i - 1]] = i
    solve(books)

File: 1073_b_vasya_and_books.cpp
/*
  Contest : 1073
  Problem B : Vasya and books
  Author : Rakesh Kumar
  Date: 12/07/2020
*/

#include <bits/stdc++.h>

int main() {
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(0);
    int n = 0;
    std::cin >> n;

    std::unordered_map<int, int> books;
    for (int i = 1; i <= n; ++i) {
        int b = 0;
        std::cin >> b;
        books[b] = i;
    }

    int top = 1;
    while (n--) {
        int b = 0;
        std::cin >> b;
        if (books[b] >= top) {
            std::cout << books[b] - top + 1 << " ";
            top = books[b] + 1;
        } else
            std::cout << "0 ";
    }
    std::cout << std::endl;

    return 0;
}
